### DISCUSSION TITLE
PHP Warning: Undefined

### DISCUSSION THREAD

**Discussion by magnusvin** [laravel/framework#46454] [Q&A]
Hello. I get warnings in syslog, f.ex in Apache logs.
```
PHP Warning:  Undefined array key \"MYSQL_ATTR_SSL_CA\" in /var/www/html/laravel/vendor/laravel/framework/src/Illuminate/Collections/Collection.php on line 1735
```
Code in Laravel:
```PHP
    /**
     * Get an item at a given offset.
     *
     * @param  TKey  $key
     * @return TValue
     */
    public function offsetGet($key): mixed
    {
        return $this->items[$key];
    }
```
I know there is a function called offsetExists(), but it does not seem to be called when using
```PHP
env('MYSQL_ATTR_SSL_CA')
```

Should i make a pull request that makes this correction:
```PHP
    /**
     * Get an item at a given offset.
     *
     * @param  TKey  $key
     * @return TValue
     */
    public function offsetGet($key): mixed
    {
        return isset($this->items[$key]) ? $this->items[$key] : null;
        // OR
        return $this->items[$key] ?? null;
        // OR
        return $this->offsetExists($key) ? $this->items[$key] : null;
    }
```
This warning comes from /config/database.php and i think ExceptionHandler cannot catch this because the application is still bootstrapping.

Please let me know if you have any other solution to this or if I am thinking wrong.

**Comment by Angel5a** [comment]
No, you should not make such a pull request. Collection still have to rise warning.

Why is your code even try to resolve MYSQL_ATTR_SSL_CA from a collection? I don't know any place that can do it. Could you provide a stack-trace to this point? For instance you can rise exception manually to get full trace in log^
```
    public function offsetGet($key): mixed
    {
        if (!isset($this->items[$key]) && $key === 'MYSQL_ATTR_SSL_CA') {
            throw new \ArgumentException('Debug break on MYSQL_ATTR_SSL_CA');
        }
        return $this->items[$key];
    }
```

**Reply by magnusvin** [reply]
If you read the whole text I wrote this on the bottom:
```
This warning comes from /config/database.php and i think ExceptionHandler cannot catch this because the application is still bootstrapping.

Please let me know if you have any other solution to this or if I am thinking wrong.
```

I am not using Collection. Laravel may convert the array to a Collection. I have not dig deep into this because this is not a big problem.

Default value is NULL and I think it should not raise a warning in this case.

**Reply by Angel5a** [reply]
This warning is comming from:

> Collection.php on line 1735

And that's it.

File /config/database.php does not trigger this error right now and has a proper extension loaded check, unless you are using something old, like Laravel 5. And there is no any hidden magic of converting array into Collections. All conversions are explicit. There are plenty of things you can do, like: сlear config cache; check your system & apache config to include all required extensions; update you setup. But for sure you should not pull any "fixes" without understanding the reason.
